src/contracts/acrossConfigStore.ts

```diff
@@ -23,8 +23,9 @@ export class Client {
   }
   static parseL1TokenConfig(data: string): L1TokenConfig {
     const l1TokenConfig = JSON.parse(data);
-    assert(l1TokenConfig, L1TokenConfigSs);
-    return l1TokenConfig;
+    const l1TokenConfigMask = mask(l1TokenConfig, L1TokenConfigSs);
+    assert(l1TokenConfigMask, L1TokenConfigSs);
```

Comment:
Something I didn't quite understand from the [mask docs](https://docs.superstructjs.org/api-reference/core#mask) was whether it guarantees that the input values contains the required properties or not. The typing of this function would make it seems like it does do the check because the returned type is T and not unknown.

If it does, it would seem that the assert isn't necessary. If not, the assert makes sense. Do you have a better understanding of how this works?

Reply:
Understood, that seems reasonable!